.github/workflows/agent-build.yml

```diff
@@ -0,0 +1,39 @@
+# The main "setup" workflow, that calls other workflows.
+name: Scalyr Agent Build.
+
+on: [push]
+
+# Agent image tests.
+jobs:
+  docker-json:
+    uses: scalyr/scalyr-agent-2/.github/workflows/docker-and-k8s.yml@add-new-docker-image-build
```

Comment:
Cool, this re-usable workflows / workflows as actions are pretty new functionality, right?

Reply:
Yes, it's pretty new and it also has one unresolved issue. For now, it can not properly run the re-usable workflow locally, from the same repo, so we have to specify its full path with `@<branch_name>`. The problem is that this `@<branch_name>` can not be dynamically evaluated with parameter expression and has to be hardcoded.
https://github.community/t/ref-head-in-reusable-workflows/203690. I added the verifier job that checks branch names and fails if they are don't match. Will hope that they fix that issue soon.